entrypoint.sh

```diff
@@ -43,28 +79,26 @@ cdktf synth --output "$ER_OUTDIR"
 $TERRAFORM_CMD init
 
 if [[ $ACTION == "Apply" ]]; then
-    if [[ $DRY_RUN == "True" ]]; then
-        $TERRAFORM_CMD plan -out=plan --lock=false
-        if [ -f "validate_plan.py" ]; then
-            $TERRAFORM_CMD show -json "$CDKTF_OUT_DIR"/plan > "$CDKTF_OUT_DIR"/plan.json
-            python3 validate_plan.py "$CDKTF_OUT_DIR"/plan.json
-        fi
-    elif [[ $DRY_RUN == "False" ]]; then
+    $TERRAFORM_CMD plan -out=plan --lock=false
+    $TERRAFORM_CMD show -json "$CDKTF_OUT_DIR"/plan > "$CDKTF_OUT_DIR"/plan.json
+    run_hook "validate_plan" "$CDKTF_OUT_DIR"/plan.json
+
+    if [[ $DRY_RUN == "False" ]]; then
         # cdktf apply isn't reliable for now, using terraform apply instead
-        $TERRAFORM_CMD apply -auto-approve
+        $TERRAFORM_CMD apply -auto-approve "$CDKTF_OUT_DIR"/plan
         $TERRAFORM_CMD output -json > "$OUTPUT_FILE"
-        if [ -f "post_checks.py" ]; then
-            python3 post_checks.py "$OUTPUT_FILE"
-        fi
+        run_hook "check_output" "$OUTPUT_FILE"
     fi
+    run_hook "post_apply" "$CDKTF_OUT_DIR"/plan.json
```

Comment:
Do we have potential scenarios in which we need to support some kind of cleanup in a try/catch manner? Like having some `on_error` hook that we try to call if anything fails in a previous stage?

Reply:
There isn't any need for that currently, but it should be pretty easy to implement if needed.